libraries/plugins/persistent_proposals/persistent_proposals_api.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2015 Blockchain Projects BV, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <graphene/persistent_proposals/persistent_proposals_api.hpp>
+
+#include <graphene/persistent_proposals/persistent_proposals_plugin.hpp>
+#include <graphene/persistent_proposals/persistent_proposals_objects.hpp>
+
+#include <graphene/chain/database.hpp>
+#include <graphene/app/application.hpp>
+
+namespace graphene { namespace persistent_proposals {
+
+namespace detail {
+
+class persistent_proposals_api_impl
+{
+public:
+    persistent_proposals_api_impl( graphene::app::application& app )
+    : _database(app.chain_database().get())
+    {}
+
+    vector<proposal_object> get_proposed_transactions( object_id_type start, unsigned int limit )const
+    {
+        return get_proposed_transactions_impl(start, limit, [](const proposal_object&){return true;});
+    }
+
+    vector<proposal_object> get_proposed_transactions_for_account( account_id_type account_id, object_id_type start, unsigned int limit )const
+    {
+        return get_proposed_transactions_impl(start, limit,
+                                              [&](const proposal_object& proposal) {
+                                                  return does_proposal_belong_to_account(proposal, account_id);
+                                              });
+    }
+
+    vector<proposal_update_object> get_proposal_updates( object_id_type proposal_id, object_id_type start, unsigned limit )const
+    {
+        FC_ASSERT(limit <= 100, "You may request at most 100 proposal updates at a time.");
```

Comment:
Since the result will contain `start`, I think 101 would be better than 100 for pagination.

Reply:
I'd like there to be 100 items in the returned list. In the while loop, it says:
`proposal_updates.size() < limit`. Maybe this should be `<=` instead, no?